src/coins.cpp

```diff
@@ -203,12 +203,29 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlockIn
 }
 
 bool CCoinsViewCache::Flush() {
-    bool fOk = base->BatchWrite(cacheCoins, hashBlock);
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase*/ true);
     cacheCoins.clear();
     cachedCoinsUsage = 0;
     return fOk;
 }
 
+bool CCoinsViewCache::Sync()
+{
+    bool fOk = base->BatchWrite(cacheCoins, hashBlock, /*erase*/ false);
+    // Instead of clearing the cache, just clear the FRESH/DIRTY
```

Comment:
nit: If I look at this comment outside of the context of this PR "Instead of clearing the cache" might not make much sense because there is no cache clearing going on inside of this function and the relation to `Flush()` may not be immediately clear.

Reply:
rephrased